execute/transport_test.go

```diff
@@ -83,3 +84,81 @@ func TestProcessMsg(t *testing.T) {
 	}
 	mem.AssertSize(t, 0)
 }
+
+func TestProcessChunkMsg(t *testing.T) {
+	fromTableChunk := func(chunk table.Chunk) flux.Table {
+		buffer := chunk.Buffer()
+		buffer.Retain()
+		return table.FromBuffer(&buffer)
+	}
+	mem := memory.NewCheckedAllocator(memory.DefaultAllocator)
+	defer mem.AssertSize(t, 0)
+
+	// Create a set of tables, wrap them in a process message,
+	// and then ack it.
+	ti := static.Table{
+		static.StringKey("_measurement", "m0"),
+		static.Times("_time", 0, 10, 20),
+		static.Floats("_value", 0, 1, 2),
+	}
+	if err := ti.Do(func(tbl flux.Table) error {
+		return tbl.Do(func(cr flux.ColReader) error {
+			chunk := table.ChunkFromReader(cr)
+			chunk.Retain()
+			m := execute.NewProcessChunkMsg(chunk)
+
+			cr.Retain()
+			want := table.Iterator{table.FromBuffer(cr)}
+			got := table.Iterator{fromTableChunk(m.TableChunk())}
+			if diff := table.Diff(want, got); diff != "" {
+				t.Errorf("unexpected table data -want/+got:\n%s", diff)
+			}
+			m.Ack()
+			return nil
+		})
+	}); err != nil {
+		t.Errorf("unexpected error: %s", err)
+	}
+	mem.AssertSize(t, 0)
```

Comment:
Both `Retain` and `Release` are just incrementing and decrementing a reference count under the hood, correct?

So in this test, you are incrementing the count with `chunk.Retain()` at line 107, and then the reference count is going back to zero at line 116, so the allocated memory should go back to zero. Do I have that right?

Reply:
Yes.